src/Elastic.Markdown/Slices/Directives/Image.cshtml

```diff
@@ -1,5 +1,20 @@
 @inherits RazorSlice<ImageViewModel>
-<a class="reference internal image-reference" href="@Model.ImageUrl">
+<a class="reference internal image-reference" href="javascript:void(0)" onclick="document.getElementById('modal-@Model.GetHashCode()').style.display='flex'">
 	<img loading="lazy" alt="@Model.Alt" src="@Model.ImageUrl" style="@Model.Style" class="@Model.Screenshot" />
 	[CONTENT]
 </a>
+
+<div id="modal-@Model.GetHashCode()" class="modal"
```

Comment:
`Guid.NewGuid().ToString("N")[..8]` instead of `GetHashCode()` 

Other than that LGTM!